examples/taxi-ride-akka/akkastreams/src/main/scala/taxiride/akka/processor/actors/RidesBehaviour.scala

```diff
@@ -0,0 +1,51 @@
+package taxiride.akka.processor.actors
+
+import akka.actor.typed._
+import akka.actor.typed.scaladsl.Behaviors
+import taxiride.datamodel._
+
+import scala.collection.mutable.Map
+
+// Controller
+trait RidesProcessingActor
+
+// messages
+case class ProcessFare(reply: ActorRef[Option[TaxiRideFare]], fare : TaxiFare) extends RidesProcessingActor
+case class ProcessRide(reply: ActorRef[Option[TaxiRideFare]], ride : TaxiRide) extends RidesProcessingActor
+
+// Actor
+object RideShare{
+
+  private val rideState: Map[Long, TaxiRide] = Map()
+  private val fareState: Map[Long, TaxiFare] = Map()
```

Comment:
This needs more work, see further down.

Reply:
There are options here. Let me finish the first path through of Flink